tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchReindexingTests.java

```diff
@@ -178,8 +179,112 @@ public void testRecreateIndex() throws Exception
+     * @param expectingDocNameAsResult Result from verification query string is the name of the document created when true, empty result when false.
+     */
+    private void internalTestEnabledFeatures(
+        Boolean metadataIndexingEnabled,
+        Boolean contentIndexingEnabled,
+        Boolean pathIndexingEnabled,
+        String queryString,
+        Boolean expectingDocNameAsResult
+    ) throws Exception
+    {
+
+        // Initial timestamp for reindexing by date: this will save reindexing time for these tests
+        ZonedDateTime now = ZonedDateTime.now(Clock.systemUTC());
+        String testStart = DateTimeFormatter.ofPattern("yyyyMMddHHmm").format(now.minusMinutes(1));
+
+        // GIVEN
+        // Stop ElasticsearchConnector
+        AlfrescoStackInitializer.liveIndexer.stop();
+        // Create document
+        String documentName = createDocument();
+        // Delete index documents
+        cleanUpIndex();
+        // Restart ElasticsearchConnector to Index Content
         AlfrescoStackInitializer.liveIndexer.start();
 
+        // WHEN
+        // Run reindexer leaving ALFRESCO_REINDEX_TO_TIME as default
+        reindex(Map.of("ALFRESCO_REINDEX_JOB_NAME", "reindexByDate",
+            "ELASTICSEARCH_INDEX_NAME", CUSTOM_ALFRESCO_INDEX,
+            "ALFRESCO_REINDEX_FROM_TIME", testStart,
+            "ALFRESCO_REINDEX_METADATAINDEXINGENABLED", metadataIndexingEnabled.toString(),
+            "ALFRESCO_REINDEX_CONTENTINDEXINGENABLED", contentIndexingEnabled.toString(),
+            "ALFRESCO_REINDEX_PATHINDEXINGENABLED", pathIndexingEnabled.toString()));
+
+        // THEN
+        if (expectingDocNameAsResult)
+        {
+            expectResultsFromQuery(queryString.replace("<DOCUMENT_NAME>", documentName), dataUser.getAdminUser(), documentName);
+        }
+        else
+        {
+            expectResultsFromQuery(queryString.replace("<DOCUMENT_NAME>", documentName), dataUser.getAdminUser());
+        }
+
+        // TIDY
+        // Delete index documents
+        cleanUpIndex();
+
+    }
+
+    @Test(groups = TestGroup.SEARCH)
+    public void testRecreateIndexWithMetadataAndContent() throws Exception
+    {
+        internalTestEnabledFeatures(true, true, false,
+            "cm:name:'<DOCUMENT_NAME>' AND TEXT:'content'", true);
+    }
+
+    @Test(groups = TestGroup.SEARCH)
+    public void testRecreateIndexWithMetadataAndNoContent() throws Exception
+    {
+        internalTestEnabledFeatures(true, false, false,
+            "cm:name:'<DOCUMENT_NAME>' AND TEXT:'content'", false);
+    }
+
+    @Test(groups = TestGroup.SEARCH)
+    public void testRecreateIndexWithNoMetadataAndContent() throws Exception
+    {
+        // When not using metadata, document shouldn't be present in Elasticsearch index,
+        // since metadata reindexing process is indexing also permissions
+        internalTestEnabledFeatures(false, true, false,
+            "cm:name:'<DOCUMENT_NAME>'", false);
```

Comment:
Do we know that this query won't hit the DB?

Reply:
We need to add `AND cm:name:*` to avoid that. Thanks!